sktime/clustering/partitioning/_averaging_metrics.py

```diff
@@ -0,0 +1,220 @@
+            Series to get the mean of
+
+        n_iterations: int
+            Number of iterations to refine the average
+        """
+        super(MeanAveraging, self).__init__(series, n_iterations)
+
+    def average(self):
+        """
+        Method used to get the mean average
+
+        Returns
+        -------
+        Numpy_Array:
+            Array containing the estimated average
+        """
+        return self.series.mean(axis=0)
+
+
+class BarycenterAveraging(BaseClusterAverage):
+    """
+    /*******************************************************************************
+     * Copyright (C) 2018 Francois Petitjean
+     *
+     * This program is free software: you can redistribute it and/or modify
+     * it under the terms of the GNU General Public License as published by
+     * the Free Software Foundation, version 3 of the License.
+     *
+     * This program is distributed in the hope that it will be useful,
+     * but WITHOUT ANY WARRANTY; without even the implied warranty of
+     * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+     * GNU General Public License for more details.
+     *
+     * You should have received a copy of the GNU General Public License
+     * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+     ******************************************************************************/
+    TODO: This is basically copied from
+    https://github.com/fpetitjean/DBA/blob/master/DBA.py
+    This need massive optimisations and likely needs to be done in
+    cython. This is basically doing its own implementation of dtw
+    because it needs the cost matrix from the dtw calculation
+    and the current sktime one doesnt return that so need to figure
+    that out
+    """
+
+    def __init__(self, series: Numpy_Array, n_iterations: int = 10):
+        """
+        Constructor for barycenter averaging
+
+        Parameters
+        ----------
+        series: Numpy_Array
+            The series to find the DBA average of
+
+        n_iterations: int
+            Number of iterations to refine the average
+        """
+        super(BarycenterAveraging, self).__init__(series, n_iterations)
+
+    def average(self):
+        """
+        Method used to get the barycenter average
+
+        Returns
+        -------
+        Numpy_Array:
+            Array containing the estimated average
+        """
+        max_length = reduce(max, map(len, self.series))
+
+        cost_mat = np.zeros((max_length, max_length))
+        delta_mat = np.zeros((max_length, max_length))
+        path_mat = np.zeros((max_length, max_length), dtype=np.int8)
+
+        medoid_ind = self.__approximate_medoid_index(cost_mat, delta_mat)
+        center = self.series[medoid_ind]
+
+        for _ in range(self.n_iterations):
+            center = self.__DBA_update(center, cost_mat, path_mat, delta_mat)
+
+        return center
+
+    def __approximate_medoid_index(self, cost_mat, delta_mat):
```

Comment:
same as above, here and below
```suggestion
    def _approximate_medoid_index(self, cost_mat, delta_mat):
```